packages/app/src/cli/models/extensions/specifications/editor_extension_collection.ts

```diff
@@ -17,17 +18,33 @@ const EditorExtensionCollectionSchema = BaseSchema.extend({
   type: zod.literal('editor_extension_collection'),
 })
 
+export type EditorExtensionCollectionType = zod.infer<typeof EditorExtensionCollectionSchema>
```

Comment:
If you define your schema like this:

```ts
const EditorExtensionCollectionSchema = BaseSchema.extend({
  include: zod.array(IncludeSchema).optional().default([]),
  includes: zod.array(zod.string()).optional().default([]),
  type: zod.literal('editor_extension_collection'),
}).transform((data) => {
  const includes = data.includes.map((handle) => {
    return {handle}
  })
  return {
    ...data,
    inCollection: [...includes, ...data.include],
  }
```

You can avoid having `makeExtensionsInCollection` and all the calls to it.

Reply:
You can also define validations at load-time in the schema like:

```ts
 .refine((data) => data.inCollection.length > 1, {
    message: 'There must be at least two extensions in editor extension collection',
  })
```

Reply:
Thanks! Ill try out this schema change

Reply:
Hmm just tried it again, it works!